go/tools/bzltestutil/wrap.go

```diff
@@ -128,19 +128,22 @@ func Wrap(pkg string) error {
 	cmd.Env = append(os.Environ(), "GO_TEST_WRAP=0")
 	cmd.Stderr = io.MultiWriter(os.Stderr, streamMerger.ErrW)
 	cmd.Stdout = io.MultiWriter(os.Stdout, streamMerger.OutW)
-	go func() {
-		// When the Bazel test timeout is reached, Bazel sends a SIGTERM that
-		// we need to forward to the inner process.
-		// TODO: This never triggers on Windows, even though Go should simulate
-		//  a SIGTERM when Windows asks the process to close. It is not clear
-		//  whether Bazel uses the required graceful shutdown mechanism.
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, syscall.SIGTERM)
-		<-c
-		cmd.Process.Signal(syscall.SIGTERM)
-	}()
 	streamMerger.Start()
-	err := cmd.Run()
+	err := cmd.Start()
+	if err == nil {
+		go func() {
+			// When the Bazel test timeout is reached, Bazel sends a SIGTERM that
+			// we need to forward to the inner process.
+			// TODO: This never triggers on Windows, even though Go should simulate
+			//  a SIGTERM when Windows asks the process to close. It is not clear
+			//  whether Bazel uses the required graceful shutdown mechanism.
+			c := make(chan os.Signal, 1)
```

Comment:
Could you try moving this line and the next out of the Go routine and before the call to `Start`? It's probably just a theoretical concern, but I realize that we otherwise may lose `SIGTERM`s sent to the process before the goroutine executes.

Reply:
Done